staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -213,7 +215,9 @@ func (pf *PortForwarder) forward() error {
 			listenSuccess = true
 		default:
 			if pf.errOut != nil {
+				pf.errLock.Lock()
```

Comment:
this doesn't seem to need locking since doesn't run in a goroutine

Reply:
oh wait, the switch case is confusing , isnt' it?

is it that if only one listener works and the rest fails it keeps working?
then it is not thread safe, indeed

Reply:
hmm, errOut seems is only used here, and this doesn't run in parallel, was this racing in your tests?

Reply:
I can remove it, but the thought was better safe than sorry for future changes.

Reply:
I prefer to not introduce mutex if they are not needed, increase the risk of deadlocking